miniz_oxide/src/inflate/mod.rs

```diff
@@ -167,10 +167,65 @@ fn decompress_to_vec_inner(
     }
 }
 
+/// Decompress one or more source slices from an iterator into the output slice.
+///
+/// * On success, returns the number of bytes that were written.
+/// * On failure, returns the failure status code.
+///
+/// This will fail if the output buffer is not large enough, but in that case
+/// the output buffer will still contain the partial decompression.
+///
+/// * `out` the output buffer.
+/// * `it` the iterator of input slices.
+/// * `zlib_header` if the first slice out of the iterator is expected to have a
+///   Zlib header. Otherwise the slices are assumed to be the deflate data only.
+/// * `ignore_adler32` if the adler32 checksum should be calculated or not.
+pub fn decompress_slice_iter_to_slice<'out, 'inp>(
+    out: &'out mut [u8],
+    it: impl Iterator<Item = &'inp [u8]>,
+    mut zlib_header: bool,
+    ignore_adler32: bool,
+) -> Result<usize, TINFLStatus> {
+    use self::core::inflate_flags::*;
+
+    let mut it = it.peekable();
+    let r = &mut DecompressorOxide::new();
+    let mut out_pos = 0;
+    while let Some(in_buf) = it.next() {
+        let has_more = it.peek().is_some();
+        let flags = {
+            let mut f = TINFL_FLAG_USING_NON_WRAPPING_OUTPUT_BUF;
+            if zlib_header {
+                f |= TINFL_FLAG_PARSE_ZLIB_HEADER;
+            }
+            if ignore_adler32 {
+                f |= TINFL_FLAG_IGNORE_ADLER32;
+            }
+            if has_more {
+                f |= TINFL_FLAG_HAS_MORE_INPUT;
+            }
+            f
+        };
+        let (status, _input_read, bytes_written) = decompress(r, in_buf, out, out_pos, flags);
+        zlib_header = false;
```

Comment:
I think this will end up making the zlib trailer (checksum) not be checked if it's only enabled at the start? Unless that's the intention?

Reply:
Right, the zlib header would only be at the start of the data. Once you're past the first slice there won't be another zlib header in the middle of the data.

The intent is that all slices form a *single* datastream, not one datastream per slice.

Reply:
oh oh, the trailer. I thought that was controlled by the adler flag. I will try to check which flag controls what in a few hours.

Reply:
It's controlled by the TINFL_FLAG_PARSE_ZLIB_HEADER flag. The TINFL_FLAG_COMPUTE_ADLER32 forces the value to be computed even if there is no zlib header, but in that case there won't be a checksum at the end to check against either.

Reply:
Alright, in https://github.com/Frommi/miniz_oxide/pull/112/commits/625341a336160ebffb0ab75b5053e2dcd3ba3829 I've adjusted things so that the zlib_header bool is not altered during the function call.